### ISSUE TITLE
Errors with TypeScript based React Native project

### ISSUE THREAD

**Issue by alexanderjarvis** [getsentry/sentry-react-native#704]
OS:
- [ ]  Windows
- [x]  MacOS
- [ ]  Linux

_Platform:_
- [ ]  iOS
- [ ]  Android
- [x] n/a

SDK:
- [x]  `@sentry/react-native`
- [ ]  `react-native-sentry`

`react-native` version: 0.61.2

---
I have following issue:
Errors with TypeScript based React Native project

**Steps to reproduce:**
- Add `@sentry/react-native` to Typescript React Native project.

**Actual result:**
- Actual
```
$ tsc
node_modules/@types/react-native/globals.d.ts:40:15 - error TS2300: Duplicate identifier 'FormData'.

40 declare class FormData {
                 ~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:5622:11
    5622 interface FormData {
                   ~~~~~~~~
    'FormData' was also declared here.
  node_modules/typescript/lib/lib.dom.d.ts:5632:13
    5632 declare var FormData: {
                     ~~~~~~~~
    and here

....

Found 22 errors.
```

**Expected result:**
- Result
TypeScript should compile without errors.

- Possible explanation
@sentry/react-native is extending a tsconfig.json from [sentry-javascript](https://github.com/getsentry/sentry-javascript/blob/master/packages/typescript/tsconfig.json) which specifies:
 `"lib": ["es6", "dom"],`

React Native types conflict with `dom` and there doesn't seem to be an easy way to exclude this lib from the project.

**Comment by HazAT** [comment]
Hey, so we overwrite the lib in our build tsconfig see:
https://github.com/getsentry/sentry-react-native/blob/master/tsconfig.build.json#L9

Can you please tell me how to repro this from a clean RN init project? Which template are you using?

**Comment by alexanderjarvis** [comment]
I'm using the one that's part of the react-native cli: https://github.com/react-native-community/react-native-template-typescript

**Comment by alexanderjarvis** [comment]
Also I looked into it a bit further after posting this issue and found that I couldn't find the corresponding code from GitHub in my `node_modules` (the part that specifies lib dom). `@sentry/types` is there but not `@sentry/typescript`. 

And so I think the issue might be one of the dependencies / 3rd party dependencies. There's a lot more `@types` dependencies added to the project after adding `@react-native/sentry`.

**Comment by alexanderjarvis** [comment]
Actually now that I try the latest template it seems to work:

```
npm i -g @react-native-community/cli
react-native init test --template react-native-template-typescript
cd test
yarn remove @types/react   # bug in latest template
yarn tsc  # 0 errors
yarn add @sentry/react-native
yarn tsc.  # 0 errors
```

**Comment by alexanderjarvis** [comment]
Is the issue: `node_modules/@snyk/composer-lockfile-parser/tsconfig.json`

It contains: ```"files": [
        "node_modules/typescript/lib/lib.es6.d.ts"
    ],```

**Comment by alexanderjarvis** [comment]
Ok I reproduced it. You just need to add the sentry import into `App.tsx` (in addition to steps above):

```
import * as Sentry from '@sentry/react-native';

Sentry.init({
  dsn: 'https://xxx@sentry.io/000',
});
```

It's because of this I think:

```
$ grep -r "/// <reference" node_modules/**.d.ts
...
node_modules/@sentry/utils/dist/is.d.ts:/// <reference lib="dom" />
node_modules/@sentry/utils/dist/misc.d.ts:/// <reference lib="dom" />
node_modules/@sentry/utils/esm/is.d.ts:/// <reference lib="dom" />
node_modules/@sentry/utils/esm/misc.d.ts:/// <reference lib="dom" />
...
```

**Comment by danieloprado** [comment]
```js
// tsconfig.json
{
  "compilerOptions": {
    //...
    "skipLibCheck": true, <~ add this
   //....
```

**Comment by alexanderjarvis** [comment]
@HazAT you need to remove the function from utils that references dom and only include that in the browser version of sentry (splitting the utils package in two or moving function).

**Comment by monotv** [comment]
Also happens with `1.0.6`.
```
    "react": "16.9.0",
    "react-native": "0.60.6",
```

**Comment by lisabaut** [comment]
Same is happening with these versions:

```
"@sentry/react-native": "1.0.9",
"react": "16.9.0",
"react-native": "0.61.2",
```

As soon as Sentry is included a lot of TypeScript errors occur

```
import * as Sentry from '@sentry/react-native';

Sentry.init({
  dsn: 'https://xxx@sentry.io/000',
});
```

Any updates on this?

**Comment by lobsterkatie** [comment]
This should be fixed in `1.1.0` (just released), as it now depends on a newer version of `@sentry/browser` (which itself [no longer references `dom`](https://github.com/getsentry/sentry-javascript/pull/2303)).
